cranelift/codegen/src/isa/x64/abi.rs

```diff
@@ -869,11 +956,32 @@ fn get_callee_saves(call_conv: &CallConv, regs: &Set<Writable<RealReg>>) -> Vec<
             .cloned()
             .filter(|r| is_callee_save_systemv(r.to_reg()))
             .collect(),
-        CallConv::WindowsFastcall => todo!("windows fastcall"),
+        CallConv::WindowsFastcall => regs
+            .iter()
+            .cloned()
+            .filter(|r| is_callee_save_fastcall(r.to_reg()))
+            .collect(),
         CallConv::Probestack => todo!("probestack?"),
     };
     // Sort registers for deterministic code output. We can do an unstable sort because the
     // registers will be unique (there are no dups).
     regs.sort_unstable_by_key(|r| r.to_reg().get_index());
     regs
 }
+
+fn compute_clobber_size(clobbers: &Vec<Writable<RealReg>>) -> u32 {
+    let mut clobbered_size = 0;
+    for reg in clobbers {
+        match reg.to_reg().get_class() {
+            RegClass::I64 => {
+                clobbered_size += 8;
+            }
+            RegClass::V128 => {
+                clobbered_size = (clobbered_size + 15) & !15;
```

Comment:
I've seen this type of thing enough that we may want to create an inline function `align` somewhere.

Reply:
Done! `align_to()` now used throughout this ABI code (and corresponding aarch64 code too).